### PR TITLE
pg_dump: dump the namespace while processing external partitions

### PR DESCRIPTION
**Pull request by adam8157**
We forgot to dump the namespace while processing external partitions, it
would be a problem since upstream pg_dump decided not to dump the
search_path, this commit fixes it.

## Here are some reminders before you submit the pull request
- [x] Add tests for the change

~_Where are the tests for pg_dump?_~

### CODE REVIEW

File: src/bin/pg_dump/pg_dump.c [hunk 1]
```diff
@@ -14537,9 +14537,10 @@ dumpTableSchema(Archive *fout, TableInfo *tbinfo)
 			for (i = 0; i < ntups; i++)
 			{
 				char tmpExtTable[500] = {0};
+				char *namespace_name = pstrdup(fmtId(tbinfo->dobj.namespace->dobj.name));
 				relname = pg_strdup(PQgetvalue(res, i, i_relname));
 				snprintf(tmpExtTable, sizeof(tmpExtTable), "%s%s", relname, EXT_PARTITION_NAME_POSTFIX);
-				appendPQExpBuffer(q, "ALTER TABLE %s ", fmtId(tbinfo->dobj.name));
+				appendPQExpBuffer(q, "ALTER TABLE %s.%s ", namespace_name, fmtId(tbinfo->dobj.name));
```
Comment (hlinnaka) [inline src/bin/pg_dump/pg_dump.c]: I'd suggest using `qualrelname`.

**Comment by hlinnaka** [comment]
> Where are the tests for pg_dump?

We don't have any tests for pg_dump as such. However, we run pg_upgrade tests after installcheck-world, which effectively runs pg_dump & restore on the 'regression' database. So you can add a test to the normal test suite at src/test/regress, that just leaves behind a table, and it will get dumped & restored as part of the pg_upgrade test.

File: src/bin/pg_dump/pg_dump.c [hunk 2]
```diff
@@ -14554,11 +14555,11 @@ dumpTableSchema(Archive *fout, TableInfo *tbinfo)
 					appendPQExpBuffer(q, "EXCHANGE PARTITION %s ",
 									  fmtId(PQgetvalue(res, i, i_parname)));
 				}
-				appendPQExpBuffer(q, "WITH TABLE %s WITHOUT VALIDATION; ", fmtId(tmpExtTable));
+				appendPQExpBuffer(q, "WITH TABLE %s.%s WITHOUT VALIDATION; ", namespace_name, fmtId(tmpExtTable));
```
Comment (baobao0206) [inline src/bin/pg_dump/pg_dump.c]: I think use function 'fmtQualifiedId' to make the full relname is a better choice.  It considers remoteVersion and determines whether namespace is required based on remoteVersion.

Comment (adam8157) [reply src/bin/pg_dump/pg_dump.c]: Good point, updated.

**Pull request merged**